src/main/java/com/pinterest/secor/common/FileRegistry.java

```diff
@@ -103,6 +103,9 @@ public FileWriter getOrCreateWriter(LogFilePath path, CompressionCodec codec)
             mWriters.put(path, writer);
             mCreationTimes.put(path, System.currentTimeMillis() / 1000L);
             LOG.debug("created writer for path {}", path.getLogFilePath());
+            LOG.debug("Register deleteOnExit for path {}", path.getLogFilePath());
+            FileUtil.deleteOnExit(path.getLogFilePath());
```

Comment:
Why do we do it at the individual file level rather than dir level? Each consumer is writing to a dir composed of PID and thread id, so those dirs are specific to a given instance.